pkg/gatewayserver/gatewayserver.go

```diff
@@ -638,6 +644,74 @@ func (gs *GatewayServer) handleUpstream(conn connectionEntry) {
 	}
 }
 
+// UpdateConnectionStats updates the stats for a single gateway connection.
+func (gs *GatewayServer) UpdateConnectionStats(ctx context.Context, conn *io.Connection) error {
+	stats := &ttnpb.GatewayConnectionStats{}
+	ct := conn.ConnectTime()
+	stats.ConnectedAt = &ct
+	stats.Protocol = conn.Frontend().Protocol()
+	if s, t, ok := conn.StatusStats(); ok {
+		stats.LastStatusReceivedAt = &t
+		stats.LastStatus = s
+	}
+	if c, t, ok := conn.UpStats(); ok {
+		stats.LastUplinkReceivedAt = &t
+		stats.UplinkCount = c
+	}
+	if c, t, ok := conn.DownStats(); ok {
+		stats.LastDownlinkReceivedAt = &t
+		stats.DownlinkCount = c
+	}
+	if min, max, median, count := conn.RTTStats(); count > 0 {
+		stats.RoundTripTimes = &ttnpb.GatewayConnectionStats_RoundTripTimes{
+			Min:    min,
+			Max:    max,
+			Median: median,
+			Count:  uint32(count),
+		}
+	}
+
+	return gs.statsRegistry.Set(ctx, conn.Gateway().GatewayIdentifiers, stats)
+}
+
+func (gs *GatewayServer) connStatsUpdater(conn connectionEntry) {
+	ctx := conn.Context()
+	uid := unique.ID(ctx, conn.Gateway())
+
+	logger := log.FromContext(ctx).WithFields(log.Fields(
+		"protocol", conn.Frontend().Protocol(),
+		"gateway_uid", uid,
+	))
+
+	defer func() {
+		logger.Debug("Delete connection stats")
+		err := gs.statsRegistry.Set(ctx, conn.Gateway().GatewayIdentifiers, nil)
+		if err != nil {
+			logger.WithError(err).Error("Failed to delete connection stats")
+		}
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-conn.StatsChanged():
+			logger.Debug("Update connection stats")
```

Comment:
Even for debug logs this is quite chatty I'm afraid, if this happens every 3 seconds for each gateway connection

Reply:
Was in for initial testing, removed